src/PureCloud.Client/Apis/WorkforceManagementApi.cs

```diff
@@ -0,0 +1,661 @@
+    {
+        ArgumentException.ThrowIfNullOrEmpty(managementUnitId);
+        ArgumentException.ThrowIfNullOrEmpty(workPlanId);
+
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+        var response = await client.DeleteAsync($"api/v2/workforcemanagement/managementunits/{Uri.EscapeDataString(managementUnitId)}/workplans/{Uri.EscapeDataString(workPlanId)}", cancellationToken);
+
+        return response.IsSuccessStatusCode;
+    }
+
+    /// <inheritdoc />
+    public async Task<bool> DeleteManagementunitWorkplanrotationAsync(string managementUnitId, string workPlanRotationId, CancellationToken cancellationToken = default)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(managementUnitId);
+        ArgumentException.ThrowIfNullOrEmpty(workPlanRotationId);
+
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+        var response = await client.DeleteAsync($"api/v2/workforcemanagement/managementunits/{Uri.EscapeDataString(managementUnitId)}/workplanrotations/{Uri.EscapeDataString(workPlanRotationId)}", cancellationToken);
+
+        return response.IsSuccessStatusCode;
+    }
+
+    #endregion
+
+    #region Adherence Operations
+
+    /// <inheritdoc />
+    public async Task<UserScheduleAdherence[]> GetAdherenceAsync(IEnumerable<string> userIds, CancellationToken cancellationToken = default)
+    {
+        ArgumentNullException.ThrowIfNull(userIds);
+
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+        var parameters = new NameValueCollection();
+        foreach (var id in userIds.Where(id => !string.IsNullOrEmpty(id)))
+        {
+            parameters.Add("userId", UriHelper.ParameterToString(id));
+        }
+
+        if (parameters.Count == 0)
+        {
+            throw new ArgumentException("At least one non-null and non-empty user ID is required", nameof(userIds));
+        }
+
+        var requestUri = UriHelper.GetUri("api/v2/workforcemanagement/adherence", parameters);
+        var response = await client.GetAsync(requestUri, cancellationToken);
+        response.EnsureSuccessStatusCode();
+
+        return await response.Content.ReadFromJsonAsync<UserScheduleAdherence[]>(_options.JsonSerializerOptions, cancellationToken);
+    }
+
+    /// <inheritdoc />
+    public async Task<QueryAdherenceExplanationsResponse> CreateAdherenceExplanationsQueryAsync(AgentQueryAdherenceExplanationsRequest body, CancellationToken cancellationToken = default)
+    {
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+        var response = await client.PostAsJsonAsync("api/v2/workforcemanagement/adherence/explanations/query", body, _options.JsonSerializerOptions, cancellationToken);
+        response.EnsureSuccessStatusCode();
```

Comment:
Again, add a new line before each `return` line.
```suggestion

        response.EnsureSuccessStatusCode();
        
```